base/server/python/pki/server/__init__.py

```diff
@@ -246,6 +246,48 @@ def restart(self):
         logger.debug('Command: %s', ' '.join(cmd))
         subprocess.check_call(cmd)
 
+    def run(self, as_current_user=False):
+
+        java_home = os.getenv('JAVA_HOME')
+
+        classpath = [
+            Tomcat.SHARE_DIR + '/bin/bootstrap.jar',
+            Tomcat.SHARE_DIR + '/bin/tomcat-juli.jar',
+            '/usr/lib/java/commons-daemon.jar'
+        ]
+
+        cmd = []
+
+        # by default run PKI server as systemd user
+        if not as_current_user:
+
+            # switch to systemd user if different from current user
+            username = pwd.getpwuid(os.getuid()).pw_name
+            if username != self.user:
+                cmd.extend(['sudo', '-u', self.user])
+
+        cmd.extend([
+            java_home + '/bin/java',
+            '-Djavax.sql.DataSource.Factory=org.apache.commons.dbcp.BasicDataSourceFactory',
```

Comment:
Not sure if you want to take this as a RFE for a separate PR, but it'd be nice to have a way to pass custom arguments here.

I notice you have a `--debug` mode which could be renamed to `--debug-log` mode; I'd like a way to start this with a JDB setup (which requires passing custom arguments).

I'm not sure if you want to make that a `--debug <port>` or just allow passing custom arguments (environment variable). The downside of the latter is you'll need to convert `cmd` from a `list` to a `str`.

Reply:
This command is trying to mimic the command executed when we run Tomcat service in the background. There is a way to customize Tomcat command with JAVA_OPTS var in tomcat.conf, but that is not implemented yet in this patch (since it requires some parsing).

Usually JDB params would be specified in JAVA_OPTS, but we could certainly make it easier by adding CLI parameters. I'd suggest keeping `--debug` as written in the patch since it has been used that way quite consistently in other commands, but we can add something like `--jdb-port <port>`.

Reply:
Hmm.. not sure. I think Tomcat would parse JAVA_OPTS from tomcat.conf (or /etc/sysconfig/tomcat) and include it as part of the command line. It does not read JAVA_OPTS env var. I'm not sure how to include JAVA_OPTS into the `cmd` array without parsing it. As you said, we might be able to use `str` instead, but making sure the args are quoted properly might be a bit complex.

Reply:
So my suggestion is to add `--jdb-port <port>` for now, and leave JAVA_OPTS for future improvements.

Reply:
Please take a look at the new patches. I added a code to do a simple parsing for JAVA_OPTS. This is probably sufficient for now.